File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayDeque<Integer> stack = new ArrayDeque<>();
        
        for(int i : arr) {
            if(stack.isEmpty()) stack.addLast(i);
            if(stack.peekLast() == i) continue;
            stack.addLast(i);
        }
        
        int len = stack.size();
        int[] answer = new int[len];
        for(int i = 0; i < len; i++) {
            answer[i] = stack.removeFirst();
        }

        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    
    answer = []
    for x in arr:
        size = len(answer)
        if size == 0: 
            answer.append(x)
            continue
        
        if answer[size-1] == x:
            continue
        
        answer.append(x)
        
    return answer